Code/GraphMol/RGroupDecomposition/RGroupScore.cpp

```diff
@@ -31,54 +31,73 @@ double score(const std::vector<size_t> &permutation,
 #ifdef DEBUG
     std::cerr << "Label: " << l << std::endl;
 #endif
-    std::map<std::string, unsigned int> matchSet;
+    std::vector<std::map<std::string, unsigned int>> matchSetVect;
     std::map<std::set<int>, int> linkerMatchSet;
 
     for (size_t m = 0; m < permutation.size(); ++m) {  // for each molecule
       auto rg = matches[m][permutation[m]].rgroups.find(l);
-      if (rg != matches[m][permutation[m]].rgroups.end()) {
+      if (rg == matches[m][permutation[m]].rgroups.end()) {
+        continue;
+      }
+      if (rg->second->is_linker) {
+        ++linkerMatchSet[rg->second->attachments];
 #ifdef DEBUG
         std::cerr << "  combined: " << MolToSmiles(*rg->second->combinedMol)
                   << std::endl;
         std::cerr << " RGroup: " << rg->second->smiles << " "
-                  << rg->second->is_hydrogen << std::endl;;
+                  << rg->second->is_hydrogen << std::endl;
+        ;
 #endif
-        unsigned int &count = matchSet[rg->second->smiles];
-        ++count;
+      }
 #ifdef DEBUG
-        std::cerr << " " << rg->second->combinedMol->getNumAtoms(false)
-                  << " score: " << count << std::endl;
+      std::cerr << " " << rg->second->combinedMol->getNumAtoms(false)
+                << " score: " << count << std::endl;
 #endif
-        if (rg->second->is_linker) {
-          ++linkerMatchSet[rg->second->attachments];
+      for (size_t posStart = 0, i = 0; posStart < rg->second->smiles.size(); ++i) {
```

Comment:
Can't you just iterate through smilesVect?

Reply:
This would be even simpler:

```
for(auto &smiles : rg->second->smilesVect) {
}
```

Reply:
Man, this was really daft: first concatenate with dots and then tokenize again on dots ;-) Thanks for catching it!